docs/dev/tutorials/cross-chain-tutorial.md

```diff
@@ -145,54 +187,98 @@ async function main() {
 main().catch((error) => {
   console.error(error);
   process.exitCode = 1;
 });
 ```
 
-5. Compile the contract and run the deployment script with:
+4. From the `L1-governance` folder root, compile and deploy the contract:
+
+::: code-tabs
+@tab npm
+```sh
+# compile contract
+npx hardhat compile
 
+# deploy contract
+npx hardhat run --network goerli ./scripts/deploy.ts
 ```
+@tab yarn
+```sh
 # compile contract
 yarn hardhat compile
 
 # deploy contract
 yarn hardhat run --network goerli ./scripts/deploy.ts
-
 ```
+:::
 
-The last command will print the deployed governance smart contract address in the terminal.
+You should see output like this:
+
+```sh
+Governance contract was successfully deployed at 0xf28Df77fa8ff56cA3084bd11c1CAF5033A7b8C4A
+```
 
 ## L2 counter
 
-Now that we have the L1 governance contract addressed, let's proceed with the counter contract on L2.
+Now that we have an address for the L1 governance contract, we can build, deploy, and test the counter contract on L2.
 
-1. To initialise the project in the `/L2-counter` folder run the following commands:
+1. `cd` into `/L2-counter` and initialize the project:
 
+::: code-tabs
+@tab npm
+```sh
+npm init -y
 ```
+@tab yarn
+```sh
 yarn init -y
-# install all dependencies
-yarn add -D typescript ts-node ethers@^5.7.2 zksync-web3@^0.13.1 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy
 ```
+:::
 
-::: tip
+2. Install the dependencies:
+
+::: code-tabs
+@tab npm
+```sh
+npm i @typechain/hardhat @types/node ts-node typescript @nomiclabs/hardhat-etherscan @nomiclabs/hardhat-waffle ethereum-waffle ethers @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy @nomicfoundation/hardhat-toolbox zksync-web3
+```
+@tab yarn
+```sh
+yarn add -D typescript ts-node ethers@^5.7.2 zksync-web3 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy
+```
+:::
 
-The current version of `zksync-web3` uses `ethers v5.7.x` as a peer dependency. An update compatible with `ethers v6.x.x` will be released soon.
+3. Set up the Hardhat project configurations, selecting **Create a Typescript project** as before:
 
+::: code-tabs
+@tab npm
+```sh
+npx hardhat 
+```
+@tab yarn
+```txt
+- Step 3 is only for the `npm` flow. 
+- Do not run `npx hardhat` if you are following the `yarn` flow.
+```
+:::
+
+::: tip
+- With `npm` you can also use the zkSync CLI to scaffold a project automatically. 
+- Find [more info about the zkSync CLI here](../../api/tools/zksync-cli/).
 :::
 
-2. Create the `hardhat.config.ts` file and paste the following code there:
+4. Replace (create if necessary) `hardhat.config.ts` with the following, replacing `goerli` with the RPC URL as used in the `goerli.json` file in the L1 Governance section:
 
 ```typescript
 import "@matterlabs/hardhat-zksync-deploy";
 import "@matterlabs/hardhat-zksync-solc";
 
 module.exports = {
   zksolc: {
-    version: "1.3.1",
+    version: "1.3.6",
```

Comment:
I believe the stable version is 1.3.5. We used 1.3.6 because of the edge case error on your machine

Reply:
it worked for both flows but i can revert